### DISCUSSION TITLE
Homeseer-How to use a port besides 80

### DISCUSSION THREAD

**Discussion by PanamonC** [Arksine/moonraker#824] [Q&A] [answered]
I have [power My-Printer] set up and works extremely well to control the printer power from Klipper and KlipperScreen. If I try to use any other port, other than the standard port:80, the printer will not respond to the power buttons in Klipper or KlipperScreen. The new port numbers, for example, 192.1.168123:12345, all work fine with browsers but make the buttons in Klipper and KS unresponsive. 

Here is my moonraker.conf power portion:

[power My_Printer]
type: homeseer
protocol: HTTP
address: 192.168.1.123
port: 12345
device: 123
user: user
password: password
on_when_job_queued: True
restart_klipper_when_powered: True
restart_delay: 1
initial_state: on
off_when_shutdown: true
off_when_shutdown_delay: 2
locked_while_printing: True

I also tried adding the port# to the end of the IP address, for example 192.168.1.123:12345
I have also made sure to port-forward the ports I've tried to my Homeseer server.

Any help would be appreciated, as port 80 has seen hundreds of unauthorized login attempts in the past week.

**Comment by Arksine** [comment, marked as answer]
Hi.  The original Homeseer contribution didn't include support for custom ports.  Its a relatively simple patch to add support, so I just pushed it in f0ae51792fd3d1b9c1141548bd61f5e0ec36f410.  I can't test this myself, however it should work as you intend if you pull the latest version of Moonraker and specifying the `port` option as you did in your example.

**Reply by PanamonC** [reply]
I just tested it, and Homeseer port selection now works perfectly. 

Thank you very much for the quick response!
